src/com/owncloud/android/ui/preview/PrepareVideoPlayerAsyncTask.java

```diff
@@ -0,0 +1,170 @@
+package com.owncloud.android.ui.preview;
+
+import android.accounts.Account;
+import android.accounts.AuthenticatorException;
+import android.accounts.OperationCanceledException;
+import android.app.Activity;
+import android.content.Context;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Handler;
+import android.util.Base64;
+
+import com.google.android.exoplayer2.extractor.DefaultExtractorsFactory;
+import com.google.android.exoplayer2.source.ExtractorMediaSource;
+import com.google.android.exoplayer2.source.MediaSource;
+import com.google.android.exoplayer2.upstream.DataSource;
+import com.google.android.exoplayer2.upstream.DefaultBandwidthMeter;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.google.android.exoplayer2.upstream.DefaultHttpDataSourceFactory;
+import com.google.android.exoplayer2.upstream.HttpDataSource;
+import com.owncloud.android.MainApp;
+import com.owncloud.android.datamodel.OCFile;
+import com.owncloud.android.lib.common.OwnCloudBasicCredentials;
+import com.owncloud.android.lib.common.OwnCloudCredentials;
+import com.owncloud.android.lib.common.OwnCloudSamlSsoCredentials;
+import com.owncloud.android.lib.common.accounts.AccountUtils;
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Task for prepare video player asynchronously
+ */
+public class PrepareVideoPlayerAsyncTask extends AsyncTask<Object, Void, MediaSource> {
+
+    private Context mContext;
+    private final WeakReference<OnPrepareVideoPlayerTaskListener> mListener;
+    private OCFile mFile;
+    private Account mAccount;
+    private Handler mHandler;
+
+    private static final DefaultBandwidthMeter BANDWIDTH_METER = new DefaultBandwidthMeter();
+
+
+    public PrepareVideoPlayerAsyncTask(Activity activity, OCFile file, Account account,
+                                       Handler mainHandler) {
+        mContext = activity.getApplicationContext();
+        mListener = new WeakReference<>((OnPrepareVideoPlayerTaskListener)activity);
```

Comment:
I'd rather use ```OnPrepareVideoPlayerTaskListener``` as the parameter type, that would prevent the need of a casting.